src/libraries/PositionConfig.sol

```diff
@@ -3,17 +3,58 @@ pragma solidity ^0.8.24;
 
 import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
 
-// A PositionConfig is the input for creating and modifying a Position in core, set per tokenId
+// A PositionConfig is the input for creating and modifying a Position in core, whose truncated hash is set per tokenId
 struct PositionConfig {
     PoolKey poolKey;
     int24 tickLower;
     int24 tickUpper;
 }
 
-/// @notice Library for computing the configId given a PositionConfig
+/// @notice Library to get and set the PositionConfigId and subscriber status for a given tokenId
 library PositionConfigLibrary {
+    using PositionConfigLibrary for PositionConfig;
+
+    bytes32 constant MASK_UPPER_BIT = 0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    bytes32 constant DIRTY_UPPER_BIT = 0x8000000000000000000000000000000000000000000000000000000000000000;
+
+    /// @notice returns the truncated hash of the PositionConfig for a given tokenId
+    function getConfigId(mapping(uint256 => bytes32) storage positionConfigs, uint256 tokenId)
+        internal
+        view
+        returns (bytes32 configId)
+    {
+        configId = positionConfigs[tokenId] & MASK_UPPER_BIT;
+    }
+
+    function setConfigId(
```

Comment:
should this be called in `mint`? or removed? it currently isnt called anywhere

Reply:
oh ya.. I think merge conflicts got the best of me here